docs/pages/project/contributing/contributing-cli-tests.md

```diff
@@ -152,7 +158,40 @@ For consistency, we will keep the prefix *00-* for the command under test in the
 └── 01-model-list.bats
 ```
 
-### Run End-to-End (locally)
+### Test implementation
+
+`mesheryctl` provides several commands and subcommands, to be able to add test we need to understand if there are interaction and or prerequisite to implement a test using the mre effective way.
+
+When creating a test we need to have in mind the following rules:
+
+#### Test name
+
+It must follow this naming convention **mesheryctl command name and subcommand name under test the context of execution and the expected result**
+
+**Example:**
+
+```bash
+@test "mesheryctl model view without model name should display an error" {
+  ... test implementation ...
+}
+```
+
+### Test data
+
+If you are testing a command that required a specific id, name or any value that must exist to be able to run the test, the data should have been created by you or another test before. We will not rely on data we don't control because it can result in unexpected result.
+
```

Comment:
[nit] grammar 

suggestion: 
`If a command requries a specific if, name, or any predefined value ensure that the data is created by your test or another test beforehand. Do not rely on external or uncontrolled data as it will lead to unexpected results`